src/runtime/synchronization_common.h

```diff
@@ -1138,4 +1138,36 @@ WEAK void halide_cond_wait(struct halide_cond *cond, struct halide_mutex *mutex)
    fast_cond->wait(fast_mutex);
 }
 
+WEAK halide_mutex_array* halide_mutex_array_create(int sz) {
```

Comment:
Not for this PR, but if sz is huge, we should probably hash it down to something smaller in the accessors below. Is there any harm in using the same mutex for multiple buckets (e.g. risk of deadlock)?

Reply:
I don't think there's a risk of deadlock but I added some comments based on your comments.